### ISSUE TITLE
Similarly named method signature clash resulting in compile time error

### ISSUE THREAD

**Issue by TheMightyGit** [traefik/yaegi#1163]
The following program `yaegi_wrong_method.go` triggers a **compile error**:

```go
package main

import "fmt"

type WidgetEvent struct {
        Nothing string
}

type WidgetControl interface {
        HandleEvent(e *WidgetEvent)
}

type Button struct{}

func (b *Button) HandleEvent(e *WidgetEvent) {
}

type WindowEvent struct {
        Something int
}

type Window struct {
        Widget WidgetControl
}

func (w *Window) HandleEvent(e *WindowEvent) {
}

func main() {
        window := Window{
                Widget: &Button{},
        }
        windowevent := &WindowEvent{}
        // The next line uses the signature from the wrong method, resulting in an error.
        // Renaming one of the clashing method names fixes the problem.
        window.HandleEvent(windowevent)
        fmt.Println("OK!")
}
```

Expected result:
```console
$ go run ./yaegi_wrong_method.go
OK!
```

Got:
```console
$ yaegi ./yaegi_wrong_method.go
run: yaegi_wrong_method.go:36:21: cannot use type *main.WindowEvent as type
```

Error started from commit: e32b2ab6bdfc16f225868c979e565eb0340340ff

Workaround: change the name of one of the clashing methods.

**Comment by TheMightyGit** [comment]
I've noticed that if I change the test code slightly to call the method on a _reference_ of type `Window`...

```
...
window := &Window{
...
```

...I get a new error (not seen using `go run`)...

```
> yaegi ./yaegi_wrong_method.go
run: ./yaegi_wrong_method.go:36:21: cannot use type *main.WindowEvent as type
```

----------

Also, in my application code I get a different type of error - I'll see if I can work out the small repro for this but the error is of the form...

```
reflect.Value.Convert: value of type *game.Input cannot be converted to type *struct { Type int; PosArea image.Point; Po
sPixels image.Point; WheelDelta image.Point; Keys []int32 }
```

...which, given my apps code, is still comparing the wrong types from methods of the same name somehow.

**Comment by mvertes** [comment]
Do you have it with #1164 applied ?

**Comment by TheMightyGit** [comment]
> Do you have it with #1164 applied ?

Yes, this is using #1164 - sorry, I though I was commenting on the PR 🤦

**Comment by mvertes** [comment]
ok, I can reproduce it as well. It seems to be a variant of the same issue. Will amend #1164.

**Comment by TheMightyGit** [comment]
@mvertes The pointer fix (f8dac8f) in branch fix-1163 has fixed thew errors in both test cases (ptr and non-ptr) and in my app as well <3 <3 <3
